app/controllers/admin/works_controller.rb

```diff
@@ -119,7 +121,42 @@ def download_ohms_xml
     send_data @work.oral_history_content!.ohms_xml_text,
       :type => 'text/xml; charset=UTF-8;',
       :disposition => ContentDisposition.format(disposition: "attachment", filename: "#{@work.oral_history_content!.ohms_xml.accession}.xml")
-      #:disposition => "attachment; filename=#{@work.oral_history_content!.ohms_xml.accession}.xml"
+  end
+
+  # PATCH/PUT /admin/works/ab2323ac/submit_ohms_xml
+  def submit_searchable_transcript_source
+    unless params[:searchable_transcript_source].present?
+      redirect_to admin_work_path(@work, anchor: "nav-oral-histories"), flash: { error: "No file received" }
+      return
+    end
+    transcript = params[:searchable_transcript_source].read
+    # Very basic check for now. Not stripping HMTL tags,
+    # even though they really don't belong in these transcripts.
+    if params[:searchable_transcript_source].content_type.start_with?('text/')
+      @work.oral_history_content!.update!(searchable_transcript_source: transcript)
+      redirect_to admin_work_path(@work, anchor: "nav-oral-histories"), notice: "Full text has been updated."
+    else
+      Rails.logger.debug("Could not accept this file as a full-text transcript.  MimeMagic calls it a #{MimeMagic.by_magic(transcript).to_s}")
```

Comment:
Are you intentionally using mimemagic in the debug message, even though no longer using it in the actual check? Do we need a debug message at all, is the user error displayed enough? (In prod, we don't generally output debug messages anwyay).

Reply:
Nah - that was just a stray mention that I neglected to delete. Good call!